homeassistant/components/gogogate2/cover.py

```diff
@@ -141,12 +141,18 @@ def supported_features(self):
 
     async def async_open_cover(self, **kwargs):
         """Open the door."""
+        if not self.is_closed:
```

Comment:
This relies on HA's state, instead of the API.

Since it is calling the API to open the door, is there a problem to call it when the door is already open?

I'm kinda missing the reasoning behind this change, hence my question.

Reply:
Use case: lets say the for is closed. You call the close cover service. Hass will show this entity as closing even though nothing is happening, because we simulate the closing status. This is even more evident when the cover is shown on a dashboard. The closing animation will be running but nothing is happening. It's confusing for the user.
This PR skips the status change so these intermediate statuses (closing, opening) are only used when needed.

Reply:
Your hardware/API do not support the transition, to fake that on HA is not allowed by design. If you remove your fake transition logic, all will work without issues.